### DISCUSSION TITLE
RE: Arctic magnetometer problems - possible solution with dual-GPS for heading has been implemented

### DISCUSSION THREAD

**Discussion by harrylerman** [dannyjacobs/ECHO#61] [Show and tell]
Link to documentation on Google docs: https://docs.google.com/document/d/1ETH5hFUutlsl2eCIkYq_p3tYC0RFm04MBbFYs4zPRR0/edit

We will do more testing in the coming weeks, particularly during an upcoming deployment to Uapishka. I will update how it goes!

The system (green is the rear 'Moving Base' connected to GPS2, red is the front 'Rover' connected to GPS1, and both units are connected to each other via their UART2 ports): 

![dronegps](https://user-images.githubusercontent.com/66405902/228353890-a5fa3739-0097-4477-9927-738dc23800da.jpg)

**Comment by dannyjacobs** [comment]
Neato! Looking really good. Some questions
1.   <1deg of error! Thats better than a magnetometer alone at lower latitudes, isn't it? 
2. Any idea how much extra mass you're adding here? Cross beams + 2 GPSes I'd guess maybe ~300g?
3. Will you still fly with a "main" GPS? I don't see one on there.
4. Are you using the Base-base for the Holybro system to get RTK positions of the drone?

**Reply by harrylerman** [reply]
1. I think it is, but it also matches what the u-blox F9P specification states for a 67cm baseline.
2. We took off the Here3 that weighs 49g, and added two 49g Holybros (not sure why 49g seems to be a magic number for drone GPS weights). The 3d printed mounts and brackets and CF tubing combined are about 80g, so the total added weight is about 130g. 
3. The main GPS is the rover. A question I'm working to answer is if now the drone's reported GPS position is the position of the rover unit, of if it is the center of the baseline. 
4. We will use the Here+ RTK base for the fixed base station. We've done only a bit of testing with that, but it seems to work ok. So there is RTK between fixed base and moving base, and then between moving base and rover.

**Reply by dannyjacobs** [reply]
Cool! It all seems to work so well!   New hardware seems to be a lot easier than the old stuff.
5. How are the flight dynamics with the added rotational angular momentum?  Did the Roll/Yaw need some tweaking?
6. Carbon tube availability is surprising!

**Reply by harrylerman** [reply]
There was still some hackiness to work out, especially since until like 3 days ago (literally, new commit to the documentation just happened), the PX4 documentation was out of date so we had to figure out which parameters to change and how to change them! Plus the wiring and the safety switch (not 100% necessary but I like the added safety factor of it).
5. Flight dynamics didn't seem affected in today's flights. But I'll keep an eye on that during future flights. The CF tubing, 3d printed mounts, and the GPS units themselves are so lightweight that I don't think they affect the drone's inertia matrix that appreciably.
6. Surprisingly painful to deal with!!
